### PR TITLE
feat(services): events management + `knex` migrations

### PR DESCRIPTION
**Pull request by andhrelja**
Includes an events/getEventTypes endpoint for frontend to POST eventType IDs instead of eventType names

### CODE REVIEW

**Comment by andhrelja** [comment]
Implement form validation using [yup](https://github.com/jquense/yup) instead of manually

**Comment by andhrelja** [comment]
Create a `quotes` table to store `get-a-quote` results. 

Question for Manuel on user activation (whether-or):
1. Once a `quote` record is approved by an Admin, the "krnji" user becomes an "active" user
2. Once an  `account` record is approved by an Admin, the "krnji" user becomes an "active" user

File: app.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ import handleError from "./source/common/middlewares/handleError.js";
 dotenv.config();
 
 import playlistManagementRoutes from "./source/playlist-management/routes.js";
+import authRoutes from "./source/auth/routes.js";
```
Comment (bikboktech) [inline app.js:10]: sort

File: app.js [hunk 2]
```diff
@@ -21,8 +23,10 @@ app.use(
 app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 
-app.use(spotifyAuth);
+// app.use(spotifyAuth);
```
Comment (bikboktech) [inline app.js]: uncomment

File: package.json [hunk 3]
```diff
@@ -15,11 +15,16 @@
   "license": "ISC",
   "dependencies": {
     "archiver": "^5.3.1",
+    "bcrypt": "^5.1.0",
     "cors": "^2.8.5",
     "deemix": "^3.6.14",
     "deezer-js": "^1.3.8",
     "dotenv": "^16.0.3",
     "express": "^4.18.2",
+    "joi": "^17.10.2",
```
Comment (bikboktech) [inline package.json]: use yup dammit

File: source/common/data/knexfile.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+// Update with your config settings.
+
+/**
+ * @type { Object.<string, import("knex").Knex.Config> }
+ */
+import * as dotenv from "dotenv";
+
+dotenv.config();
```
Comment (bikboktech) [inline source/common/data/knexfile.js]: try without dotenv

File: source/common/data/migrations/003_create_alter_entity_deps.cjs [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+      table.integer('updated_by').unsigned();
+      table
+        .foreign('updated_by')
+        .references('id')
+        .inTable('accounts');
+    })
+  }
+  const accountsIdExists = await knex.schema.hasColumn('accounts', 'id');
+  if (!accountsIdExists) {
+    knex.schema.alterTable('posts', table => {
```
Comment (bikboktech) [inline source/common/data/migrations/003_create_alter_entity_deps.cjs]: await

File: source/common/data/migrations/003_create_alter_entity_deps.cjs [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+      table.integer('account_author_id').unsigned();
+      table
+        .foreign('account_author_id')
+        .references('id')
+        .inTable('accounts');
+    })
+  }
+  const postsIdExists = await knex.schema.hasColumn('posts', 'id');
+  if (!postsIdExists) {
+    knex.schema.alterTable('posts', table => {
```
Comment (bikboktech) [inline source/common/data/migrations/003_create_alter_entity_deps.cjs]: await

File: source/common/data/migrations/003_create_alter_entity_deps.cjs [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+      table.integer('updated_by').unsigned();
+      table
+        .foreign('updated_by')
+        .references('id')
+        .inTable('accounts');
+    })
+  }
+  const eventsUpdatedByExists = await knex.schema.hasColumn('events', 'id');
+  if (!eventsUpdatedByExists) {
+    knex.schema.alterTable('events', table => {
```
Comment (bikboktech) [inline source/common/data/migrations/003_create_alter_entity_deps.cjs]: await

File: source/common/data/migrations/003_create_alter_entity_deps.cjs [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+      table
+        .foreign('post_id')
+        .references('id')
+        .inTable('posts');
+    })
+  }
+
+  const accountsUpdatedByExists = await knex.schema.hasColumn('accounts', 'id');
+  if (!accountsUpdatedByExists) {
+    knex.schema.alterTable('accounts', table => {
```
Comment (bikboktech) [inline source/common/data/migrations/003_create_alter_entity_deps.cjs]: await

**Review by bikboktech — CHANGES_REQUESTED** [review]

File: source/common/data/knexfile.js (hunk 4 above)
Comment (bikboktech) [reply source/common/data/knexfile.js]: works without dotenv

**Review by bikboktech — APPROVED** [review]

**Pull request merged**